ee/frontend/mobile-replay/transformer/status-bar.ts

```diff
@@ -19,10 +20,10 @@ function spacerDiv(idSequence: Generator<number>): serializedNodeWithId {
 /**
  * tricky: we need to accept children because that's the interface of converters, but we don't use them
  */
-export function makeStatusBar(wireframe: wireframeStatusBar, _children: serializedNodeWithId[], timestamp: number, idSequence: Generator<number>): serializedNodeWithId {
-    const clockId = idSequence.next().value;
+export function makeStatusBar(wireframe: wireframeStatusBar, _children: serializedNodeWithId[], context: ConversionContext): ConversionResult<serializedNodeWithId> {
```

Comment:
this signature change is the meat of what has gone into the PR

we now needed to pass multiple parameters around, and rely on that state persisting through chained calls

it was getting messy and confusing

so I pulled out the concept of the conversion having a context, and the steps in the conversion returning a result. 

the context can be passed along through the results 

this also opens up isolating the id  generation, and will make it easier to split this big file up a little